### PR TITLE
[language] Deprecate TransactionPayload::Program

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

This is a followup cleanup mentioned in https://developers.libra.org/blog/2019/10/22/simplifying-payloads. This PR will not be a breaking change on the validator side, but it will be a breaking change for folks using Program to submit transactions.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: language/tools/test-generation/src/lib.rs [hunk 1]
```diff
@@ -44,22 +45,12 @@ fn run_verifier(module: CompiledModule) -> Result<VerifiedModule, String> {
 
 /// This function runs a verified module in the VM runtime
 fn run_vm(module: VerifiedModule) -> Result<(), String> {
-    let modules = ::stdlib::stdlib_modules().to_vec();
```
Comment (dariorussi) [inline language/tools/test-generation/src/lib.rs]: why did you decide to go down this path? that is, change the code to `execute_function_in_module`?
Just curious and fine this way

Comment (runtian-zhou) [reply language/tools/test-generation/src/lib.rs]: I think it makes more sense to use the same code path as the rest of the functional_tests/e2e_test, during which we start the execution from the genesis state. And I think the `execute_function_in_module` is a more abstract api we can use for all the tools that we have. The idea is that we can start execute a function in a module at a given blockchain state, which is a pattern used by both the test_generation and cost_synthesis.

**Comment by runtian-zhou** [comment]
> LGTM and thanks! I'll take over again from here and continue our clean up :)
> Could you explain why you did not remove Program entirely? I mean from the proto itself and make it impossible to load one any more...
> Is the idea we would do it at a different time?

That is indeed the idea. We will remove that variant at a different time. The reason for that is because of the LCS. IIUC the enum variant number is given by the order of how each variant appeared in Enum. That is to say if we get rid of that variant, the LCS encoding for script and module would be changed which is something I don't want for now.

**Pull request closed without merging**